tests/beaker_tests/netdev_stdlib/snmp_notification_provider_defaults.rb

```diff
@@ -0,0 +1,150 @@
+# snmp_notification.
+#
+# There are 2 sections to the testcase: Setup, group of teststeps.
+# The 1st step is the Setup teststep that cleans up the switch state.
+# Steps 2-4 deal with snmp_notification resource creation and its
+# verification using Puppet Agent and the switch running-config.
+# Steps 5-7 deal with snmp_notification resource deletion and its
+# verification using Puppet Agent and the switch running-config.
+#
+# The testcode checks for exit_codes from Puppet Agent, Vegas shell and
+# Bash shell command executions. For Vegas shell and Bash shell command
+# string executions, this is the exit_code convention:
+# 0 - successful command execution, > 0 - failed command execution.
+# For Puppet Agent command string executions, this is the exit_code convention:
+# 0 - no changes have occurred, 1 - errors have occurred,
+# 2 - changes have occurred, 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+# 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+# The testcode also uses RegExp pattern matching on stdout or output IO
+# instance attributes of Result object from on() method invocation.
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+require File.expand_path('../snmp_notificationlib.rb', __FILE__)
+
+result = 'PASS'
+testheader = 'SNMP Resource :: All Attributes Defaults'
+
+# @test_name [TestCase] Executes defaults testcase for ROUTEDINTF Resource.
+test_name "TestCase :: #{testheader}" do
+  # @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider' do
+    # Define PUPPETMASTER_MANIFESTPATH constant using puppet config cmd.
+    UtilityLib.set_manifest_path(master, self)
+
+    logger.info('Setup switch for provider')
+
+    # Make sure radius server is not configured before test starts.
+    on(master, SnmpNotificationLib.create_absent)
+
+    # Expected exit_code is 0,2 since server may or may not be configured.
+    cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      'agent -t', options)
+    on(agent, cmd_str, acceptable_exit_codes: [0, 2])
+  end
+
+  step 'TestStep :: Setup switch for provider test' do
+    # Expected exit_code is 0 since this is a bash shell cmd.
+    on(master, SnmpNotificationLib.create_defaults)
+
+    # Expected exit_code is 0 since this is a puppet agent cmd with no change.
+    # Or expected exit_code is 2 since this is a puppet agent cmd with change.
+    cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      'agent -t', options)
+    on(agent, cmd_str, acceptable_exit_codes: [2])
+
+    # Expected exit_code is 0 since this is a vegas shell cmd.
+    # Flag is set to true to check for absence of RegExp pattern in stdout.
+    cmd_str = UtilityLib.get_vshell_cmd("sh run snmp all | include \"snmp-server enable traps\"")
```

Comment:
Just an FYI -- This is fine as is but we're no longer requiring "show command" tests in our beaker scripts since we don't have vsh support on the XR platforms. For future PRs you can omit these show tests. At some point we will refactor all of the tests and remove the show tests.